src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1061,10 +1064,20 @@ ModifyQuerySupported(Query *queryTree, Query *originalQuery, bool multiShardQuer
 		}
 	}
 
-	if (commandType != CMD_INSERT && multiShardQuery)
+	if (commandType != CMD_INSERT)
 	{
-		DeferredErrorMessage *errorMessage = MultiShardModifyQuerySupported(
-			originalQuery, plannerRestrictionContext);
+		DeferredErrorMessage *errorMessage = NULL;
+
+		if (multiShardQuery)
+		{
+			errorMessage = MultiShardModifyQuerySupported(originalQuery,
+														  plannerRestrictionContext);
+		}
+		else
+		{
+			errorMessage = SingleShardModifyQuerySupported(originalQuery,
```

Comment:
kind of wonder whether we should change ModifyQuery -> UpdateDelete in these names.